internal/supervisor/installation.go

```diff
@@ -1456,6 +1489,46 @@ func getAnnotationsNames(annotations []*model.Annotation) []string {
 	return names
 }
 
+func (s *InstallationSupervisor) manageInstallationResourcesCache() {
+	if !s.scheduling.balanceInstallations {
+		return
+	}
+
+	s.logger.Info("Starting installation supervisor cache manager")
+
+	// Create new logger with error level logging to keep logs clean.
+	cacheLogger := log.New().WithFields(log.Fields{
+		"instance": s.instanceID,
+		"cache":    "cluster-resources",
+	})
+	cacheLogger.Logger.SetLevel(log.ErrorLevel)
+
+	for {
+		clusters, err := s.store.GetClusters(&model.ClusterFilter{Paging: model.AllPagesNotDeleted()})
+		if err != nil {
+			cacheLogger.WithError(err).Warn("Failed to query clusters")
+		} else {
+			for _, cluster := range clusters {
+				clusterResources, err := s.provisioner.GetClusterResources(cluster, true, cacheLogger)
+				if err != nil {
+					cacheLogger.WithError(err).Error("Failed to get cluster resources")
+					continue
+				}
+				s.cache.installationMetrics[cluster.ID] = clusterResources
```

Comment:
We might need to use a mutex for the `map`. I am not sure if concurrent one write + multiple reads is thread-safe.

Reply:
Good point. I had considered concurrent reads being okay, but reading the docs confirmed that we should lock even with one writer in the mix.